backend/src/payment/providers/checkout-v3.ts

```diff
@@ -0,0 +1,209 @@
+import payment = require('../index');
+const config = require('../../../config/config').payment['checkout-v3'];
+import order = require('../../order');
+import ticket = require('../../ticket');
+import log = require('../../log');
+import { to, reject, resolve } from '../../util';
+
+import _ = require('underscore');
+import moment = require('moment');
+import uuidv4 = require('uuid/v4'); //used for nonce generation
+import crypto = require('crypto');
+import axios from 'axios';
+import express = require('express');
+
+const PROVIDER = 'checkout-v3';
+
+log.info('Loaded provider: ' + PROVIDER);
+
+export async function create(order: order.IOrder, args: payment.ICreateArgs): Promise<payment.ICreateResponse> {
+
+  var nonce = uuidv4();
+  var body = orderToCreateRequestBody(order, args);
+  var headers: CheckoutHeaders = {
+    'checkout-account': config.user,
+    'checkout-algorithm': 'sha256',
+    'checkout-method': 'POST',
+    'checkout-nonce': nonce,
+    'checkout-timestamp': moment().format()
+  }
+
+  var signature = sign(headers, body);
+
+  var httpHeaders = {
+    'content-type': 'application/json',
+    'charset': 'utf-8',
+    'signature': signature
+  }
+
+  let resp, err;
+
+  [resp, err] = await to(axios({
+      baseURL: 'https://api.checkout.fi',
+      method: 'post',
+      url: '/payments',
+      headers: httpHeaders,
+      data: body,
+      params: headers
+    }));
+
+  if (err) {
+    if (!verify(err.response.headers.signature, err.response.headers, err.response.data)) {
+          log.error('Response signature validation failed', {order_id: order.order_id});
+    }
+    log.error('Posting new payment failed', {error: {data: err.response.data, status: err.response.status, header: err.response.headers}, provider: PROVIDER});
+    return reject(err.response.data);
+  }
+
+  if (!verify(resp.headers.signature, resp.headers, resp.data)) {
+    return reject({name: 'Verification error', message: 'Signature verification failed'});
+  }
+
+  const createResp: payment.ICreateResponse = {
+    payment_id: resp.data.transactionId,
+    redirect_url: resp.data.href,
+    payment_url: resp.data.href,
+    payment_provider: PROVIDER,
+    payload: resp.data.providers
+  }
+  
+  return resolve(createResp);
+}
+
+// TODO: implement callback handlers
+export function handleSuccessCallback(req: express.Request): Promise<payment.ISuccessResponse> {
```

Comment:
I'd consider renaming these to "verifySuccess" or something, as that's what it does. It's not really a callback IMO.